examples/sampling/demo_dps.py

```diff
@@ -0,0 +1,362 @@
+`Chung et al. <https://arxiv.org/abs/2209.14687>`_ The full algorithm is implemented in
+:meth:`deepinv.sampling.DPS`.
+"""
+
+
+# %% Installing dependencies
+# -----------------------------
+# Let us ``import`` the relevant packages, and load a sample
+# image of size 64x64. This will be used as our ground truth image.
+# .. note::
+#           We work with an image of size 64x64 to reduce the computational time of this example.
+#           The algorithm works best with images of size 256x256.
+#
+
+import numpy as np
+import torch
+
+import deepinv as dinv
+from deepinv.utils.plotting import plot
+from deepinv.optim.data_fidelity import L2
+from deepinv.utils.demo import load_url_image
+from tqdm import tqdm  # to visualize progress
+
+device = dinv.utils.get_freer_gpu() if torch.cuda.is_available() else "cpu"
+
+url = (
+    "https://mycore.core-cloud.net/index.php/s/9EzDqcJxQUJKYul/"
+    "download?path=%2Fdatasets&files=butterfly.png"
+)
+x_true = load_url_image(url=url, img_size=64).to(device)
+x = x_true.clone()
+
+# %%
+# In this tutorial we consider random inpainting as the inverse problem, where the forward operator is implemented
+# in :meth:`deepinv.physics.Inpainting`. In the example that we use, 90% of the pixels will be masked out randomly,
+# and we will additionally have Additive White Gaussian Noise (AWGN) of standard deviation  12.75/255.
+
+sigma = 12.75 / 255.0  # noise level
+
+physics = dinv.physics.Inpainting(
+    tensor_size=(3, x.shape[-2], x.shape[-1]),
+    mask=0.1,
+    pixelwise=True,
+    device=device,
+)
+
+y = physics(x_true)
+
+imgs = [y, x_true]
+plot(
+    imgs,
+    titles=["measurement", "groundtruth"],
+)
+
+
+# %%
+# Diffusion model loading
+# ----------------------------
+#
+# We will take a pre-trained diffusion model that was also used for the DiffPIR algorithm, namely the one trained on
+# the FFHQ 256x256 dataset. Note that this means that the diffusion model was trained with human face images,
+# which is very different from the image that we consider in our example. Nevertheless, we will see later on that
+# ``DPS`` generalizes sufficiently well even in such case.
+
+
+model = dinv.models.DiffUNet(large_model=False).to(device)
+
+# %%
+# Define diffusion schedule
+# ----------------------------
+#
+# We will use the standard linear diffusion noise schedule. Once :math:`\beta_t` is defined to follow a linear schedule
+# that interpolates between :math:`\beta_{\rm min}` and :math:`\beta_{\rm max}`,
+# we have the following additional definitions:
+# :math:`\alpha_t := 1 - \beta_t`, :math:`\bar\alpha_t := \prod_{j=1}^t \alpha_j`.
+# The following equations will also be useful
+# later on (we always assume that :math:`\mathbf{\epsilon} \sim \mathcal{N}(\mathbf{0}, \mathbf{I})` hereafter.)
+#
+# .. math::
+#
+#           \mathbf{x}_t = \sqrt{\beta_t}\mathbf{x}_{t-1} + \sqrt{1 - \beta_t}\mathbf{\epsilon}
```

Comment:
I wonder if we want to keen \mathbf for the variables because we are usually not doing this in general in the doc (or am I wrong?).

Reply:
Although now I see that I did the same thing in the DiffPIR demo... Should we remove bf in both places?

Reply:
I had the same question...

to be honest, with the number of scalars in diffusion, I think the mathbf is handy

Reply:
agreed